core-rust/state-manager/src/state_manager.rs

```diff
@@ -209,13 +217,52 @@ where
         };
 
         execute_preview(
-            &self.store,
+            &self.staged_store.root,
             &self.scrypto_interpreter,
             &self.intent_hash_manager,
             &self.network,
             preview_intent,
         )
     }
+
+    fn execute_with_cache(
+        &mut self,
+        parent_accumulator_hash: &AccumulatorHash,
+        executable: &Executable,
+        payload_hash: &LedgerPayloadHash,
+    ) -> (AccumulatorHash, &TransactionReceipt) {
+        let new_accumulator_hash = parent_accumulator_hash.accumulate(payload_hash);
+
+        match self.execution_cache.get(&new_accumulator_hash) {
+            None => {
+                let parent_key = self
+                    .execution_cache
+                    .get(parent_accumulator_hash)
+                    .expect("Parent AccumulatorHash not found in cache. This should not happen!");
```

Comment:
When could this happen? I guess if the node tries to prepare a vertex without giving the transactions in the path from the committed state?

Reply:
Could we be a little clearer in the error message about why we believe this could never happen? (eg because of specific logic in StateComputerLedger.java or something?)